src/Checkbox.jsx

```diff
@@ -96,23 +93,17 @@ export default class Checkbox extends React.Component {
     const state = this.state;
     const prefixCls = props.prefixCls;
     let checked = state.checked;
-    if (typeof checked === 'boolean') {
-      checked = checked ? 1 : 0;
-    }
-    const className = classNames({
-      [props.className]: !!props.className,
-      [prefixCls]: 1,
+
+    const className = classNames([prefixCls], {
+      [props.className]: props.className,
       [`${prefixCls}-checked`]: checked,
-      [`${prefixCls}-checked-${checked}`]: !!checked,
```

Comment:
I agree that compatibility is important,but the only way to broke this is that user define the `rc-checkbox-checked-1` class,the probability is too tiny.